clvm/SExp.py

```diff
@@ -98,13 +98,13 @@ def _to_sexp_type(
 
                 raise ValueError("can't cast to %s: %s" % (class_, v))
             if op == 1: # set left
-                stack[target] = (stack.pop(), stack[target][1])
+                stack[target] = (CLVMObject(stack.pop()), stack[target][1])
```

Comment:
Are these conversions because the things coming off the stack might be Python tuples?

Reply:
the short answer is that this fix restores the original behavior. I understood this code a bit better after richard explained it last night; the conversions in `as_tuple()` aren't supposed to be deep, they are supposed to be short-circuited by the types already being a `CLVMObject` (or `SExp`) type. This makes sure they actually are. Otherwise the whole tree has to be traversed to wrap everything every time `as_tuple()` is called.